adage/analyze/tests.py

```diff
@@ -625,19 +642,21 @@ def test_activity_mlmodel_filter(self):
         """
         Test "activity/?sample=<id>&mlmodel=<ml_id>" API.
         """
-        uri = (self.baseURI + "activity/?sample=" +
-               str(self.random_object(Sample).id) + "&mlmodel=" +
-               str(self.random_object(MLModel).id))
+        uri = self.baseURI + "activity/"
+        data = {
```

Comment:
I replied via e-mail, but it never showed up...

This is exactly right. Those special characters I added to the above test (`#`, `&`, and `%`) are all specially treated in URLs and need to be properly escaped if they are in a parameter name or value.